extensions/lifecycle/LifecycleQueuePopulator.js

```diff
@@ -320,13 +322,87 @@ class LifecycleQueuePopulator extends QueuePopulatorExtension {
         });
     }
 
+    _handleDeleteOp(entry) {
+        const value = JSON.parse(entry.value);
+
+        const locationName = value['x-amz-storage-class'];
+        const locationConfig = this.locationConfigs[locationName];
+        if (!locationConfig) {
+            this.log.error('could not get location configuration', {
+                method: 'LifecycleQueuePopulator._handleDeleteOp',
+                location: locationName,
+                bucket: entry.bucket,
+                key: entry.key,
+            });
+            return;
+        }
+
+        // cold delete only supported for archived objects in DMF backend
+        if (locationConfig.type !== 'dmf' || !value.archive) {
+            return;
+        }
+
+        const isMaster = isMasterKey(entry.key);
+        const isVersionedMaster = isMaster && !!value.versionId && !value.isNull;
+        const isNullVersion = !isMaster && value.isNull;
+        const isDeleteMarker = value.isDeleteMarker;
+        if (isVersionedMaster || isNullVersion || isDeleteMarker) {
+            this.log.trace('skip processing of object master entry');
+            return;
+        }
+
+        this.log.trace(
+            'publishing object delete entry',
+            { bucket: entry.bucket, key: entry.key, version: value.versionId },
+        );
+
+        const topic = `${coldStorageGCTopicPrefix}${locationName}`;
+        const key = `${entry.bucket}/${entry.key}`;
+
+        let version;
+        if (value.versionId) {
+            version = encode(value.versionId);
+        }
+
+        const message = JSON.stringify({
+            bucketName: entry.bucket,
+            objectKey: value.key,
+            objectVersion: version,
+            archiveInfo: value.archive.archiveInfo,
+            requestId: uuid(),
+        });
+
+        const producer = this._producers[topic];
+        if (producer) {
+            const kafkaEntry = { key: encodeURIComponent(key), message };
+            producer.send([kafkaEntry], err => {
+                LifecycleMetrics.onKafkaPublish(this.log, 'ColdStorageGCTopic', 'queuePopulator', err, 1);
+                if (err) {
+                    this.log.error('error publishing object delete request entry', {
+                        error: err,
+                        method: 'LifecycleQueuePopulator._handleDeleteOp',
+                    });
+                }
+            });
+        } else {
+            this.log.error(`producer not available for location ${locationName}`, {
+                method: 'LifecycleQueuePopulator._handleDeleteOp',
+            });
```

Comment:
shall we log the topic here as well, to ease debugging if anything happens?

Reply:
I'm thinking it's not necessary, as there is only one deletion topic per cold location, so having only the location name should be good for debugging.